src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -1,41 +1,126 @@
+      String name) {
+    return new CreateCollectionOperation(
+        commandContext, objectMapper, schemaManager, name, false, 0, null, null);
   }
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
+    return schemaManager
+        .getTable(commandContext.namespace(), name, MISSING_KEYSPACE_FUNCTION)
+        .onItemOrFailure()
+        .transformToUni(
+            (table, error) -> {
+              // if getTable return null, continue
+              if (error != null) {
+                return executeQuery(queryExecutor);
+              }
+              // table doesn't exist
+              if (table == null) {
+                return executeQuery(queryExecutor);
+              }
+              // if table exists and user want to create a vector collection with the same name
+              if (vectorSearch) {
+                // get collection settings from the existing collection
+                CollectionSettings collectionSettings =
+                    CollectionSettings.getCollectionSettings(table, objectMapper);
+                // if existing collection is a vector collection
+                if (collectionSettings.vectorEnabled()) {
+                  // get collection settings from user input
+                  CollectionSettings collectionSettings_cur =
+                      CollectionSettings.getCollectionSettings(
+                          name,
+                          vectorSearch,
+                          vectorSize,
+                          CollectionSettings.SimilarityFunction.fromString(vectorFunction),
+                          vectorize,
+                          objectMapper);
+                  if (collectionSettings.equals(collectionSettings_cur)) {
+                    // if settings are equal, no error
+                    return executeQuery(queryExecutor);
+                  } else {
+                    // if settings are not equal, error out
+                    return Uni.createFrom()
+                        .failure(
+                            new JsonApiException(
+                                ErrorCode.INVALID_COLLECTION_NAME,
+                                "The provided collection already exists with a different vector setting"));
+                  }
+                } else {
+                  // if existing collection is a non-vector collection, error out
+                  return Uni.createFrom()
+                      .failure(
+                          new JsonApiException(
+                              ErrorCode.INVALID_COLLECTION_NAME,
+                              "The provided collection already exists with a non-vector setting"));
+                }
+              }
+              // if table exists and user want to create a non-vector collection, continue
+              return executeQuery(queryExecutor);
+            });
+  }
+
+  private Uni<Supplier<CommandResult>> executeQuery(QueryExecutor queryExecutor) {
```

Comment:
If this is not just generic query execution, maybe name differently: this tries to do the actual table creation, right? If so, something like `executeTableCreation()` or somehting.